beacon-chain/rpc/eth/beacon/pool.go

```diff
@@ -306,8 +307,61 @@ func (bs *Server) SubmitVoluntaryExit(ctx context.Context, req *ethpbv1.SignedVo
 	return &emptypb.Empty{}, nil
 }
 
+// SubmitSignedBLSToExecutionChanges submits said object to the node's pool
+// if it passes validation the node must broadcast it to the network.
+func (bs *Server) SubmitSignedBLSToExecutionChanges(ctx context.Context, req *ethpbv2.SubmitBLSToExecutionChangesRequest) (*emptypb.Empty, error) {
+	ctx, span := trace.StartSpan(ctx, "beacon.SubmitVoluntaryExit")
+	defer span.End()
+	st, err := bs.ChainInfoFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+	var failures []*helpers.SingleIndexedVerificationFailure
+	for i, change := range req.GetChanges() {
+		alphaChange := migration.V2SignedBLSToExecutionChangeToV1Alpha1(change)
+		_, err = blocks.ValidateBLSToExecutionChange(st, alphaChange)
+		if err != nil {
+			failures = append(failures, &helpers.SingleIndexedVerificationFailure{
+				Index:   i,
+				Message: "Could not validate SignedBLSToExecutionChange: " + err.Error(),
+			})
+			continue
+		}
+		if err := blocks.VerifyBLSChangeSignature(st, change); err != nil {
+			failures = append(failures, &helpers.SingleIndexedVerificationFailure{
+				Index:   i,
+				Message: "Could not validate signature: " + err.Error(),
+			})
+			continue
+		}
+		bs.BLSChangesPool.InsertBLSToExecChange(alphaChange)
+		if st.Version() >= version.Capella {
+			if err := bs.Broadcaster.Broadcast(ctx, alphaChange); err != nil {
```

Comment:
This broadcast feels a bit silent. I wonder if we can log something at the end that we successfully inserted indices x, y, z into the pool and broadcast them, but failed indices a, b, c and neither inserted nor broadcast

Reply:
For our users we will provide a verbose version in prysmctl that will log everything detailed. A curl to the endpoint will only log on errors. I feel it's a good compromise cause otherwise the UX with prysmctl will log everything twice